### PR TITLE
[9.0][port] stock_split_picking

### PR DESCRIPTION
**Pull request by yvaucher**
TODO: fix tests

### CODE REVIEW

File: stock_split_picking/README.rst [hunk 1]
```diff
@@ -40,6 +63,8 @@ Maintainer
 
 This module is maintained by the OCA.
 
-OCA, or the Odoo Community Association, is a nonprofit organization whose mission is to support the collaborative development of Odoo features and promote its widespread use.
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
 
 To contribute to this module, please visit http://odoo-community.org.
```
Comment (elicoidal) [inline stock_split_picking/README.rst]: https

File: stock_split_picking/README.rst [hunk 2]
```diff
@@ -18,8 +20,29 @@ This module only needs `stock` module.
 Known issues / Roadmap
 ======================
 
-* When splitting a picking in an unassigned state, wizard won't
-  be auto completed with picking lines.
+* Spliting picking with packs is not implemented
+* It is not possible to the official method `do_split` of stock picking
```
Comment (elicoidal) [inline stock_split_picking/README.rst]: "It is not possible to super the official method `do_split` of stock picking"

**Comment by jgrandguillaume** [comment]
Hi,

On a functional point of view, it works like a charm :+1: 

Regards

**Comment by JordiBForgeFlow** [comment]
@Eficent Can you review @gmefcent?

File: stock_split_picking/views/stock_picking.xml [hunk 3]
```diff
@@ -7,8 +7,11 @@
       <field name="inherit_id" ref="stock.view_picking_form" />
       <field name="arch" type="xml">
         <field name ="state" position="before">
-          <button name="split_process"
-                  states="draft,confirmed,assigned,partially_available"
+          <xpath expr="//field[@name='do_prepare_partial]/..'" position="attributes">
+            <attribute name="attrs">"{'invisible': [('state', 'in', ('draft'))]}"</attribute>
```
Comment (gmeficent) [inline stock_split_picking/views/stock_picking.xml]: There is some issue with the view.
image

**Review by gmeficent — COMMENTED** [review]
Functional review on local :+1: Good work.
Maybe the picking lines ought to be automatically refreshed  when clicking 'split'.
[MIG] Next step would be porting yml tests to python and updating xml tags.

**Comment by Cedric-Pigeon** [comment]
@yvaucher Hi, do you plan to fix this PR ?

**Comment by yvaucher** [comment]
@Cedric-Pigeon I'll have a look at it today

**Comment by AaronHForgeFlow** [comment]
Hi @yvaucher Thank you for this awesome contribution :1st_place_medal:  Are you planning to work on this?

**Comment by yvaucher** [comment]
@aheficent It seems this is an issue of test isolation
stock_no_negative module is adding constraint and make the tests fail. Thus I'll try to add an entry in travis to separate test and see where it lead us.

**Comment by pedrobaeza** [comment]
It's better to inhibit stock_no_negative behavior when testing other modules. See https://github.com/OCA/bank-payment/blob/f8683dd7b0d313ebef31008224781b47cfc4b8a9/account_payment_transfer_reconcile_batch/models/payment_order.py#L28 for an example on inhibition.

**Comment by yvaucher** [comment]
Ok then no time yet to make the change inside `stock_no_negative` or search in existing PR if it was already done as I think there was already some other PR with the same issue. Maybe I can do it later this week.

**Comment by oihane** [comment]
Hi @yvaucher 
I was testing this module in a local instance because there is an issue for version 8.0 that a splitted picking writes down date_done field. I do not work with version 9.0 so maybe is that I missunderstood but here you have some screenshoots.

fireshot capture 6 - wh_out_00008 - odoo_ - http___localhost_8969_web
First one this is the new picking, you can see that the operation has 100 units (originally there were 180 units)

fireshot capture 5 - wh_out_00005 - odoo_ - http___localhost_8969_web
This one is the splitted picking, operation continues with 180 units (I think it should be 80 units)

fireshot capture 7 - wh_out_00005 - odoo_ - http___localhost_8969_web
As you can see here stock.move has only 80 units.

But as I said before I really didn't work with v9.0

**Comment by oihane** [comment]
Also, why don't you use this https://github.com/OCA/OCB/blob/9.0/addons/stock/stock.py#L1724 ?

**Comment by yvaucher** [comment]
@oihane please have a look at the readme for why we don't use do_split:

https://github.com/OCA/stock-logistics-workflow/pull/201/files#diff-38e0562143fc92864a575aa0bacc8658R24

**Comment by oihane** [comment]
OK, sorry I didn't read that I was focused on why in v8.0 it writes down date_done.

**Comment by rousseldenis** [comment]
@yvaucher There is a duplicate of this https://github.com/OCA/stock-logistics-workflow/pull/381

As it's a backport of v10, do you mind if that one superseed this ?

**Comment by yvaucher** [comment]
let's close this one in favor of #381

**Pull request closed without merging**